pandapipes/pipeflow.py

```diff
@@ -269,17 +269,22 @@ def solve_temperature(net):
     branch_pit[mask, TO_NODE_T] = branch_pit[mask, FROM_NODE]
 
     for comp in net['component_list']:
-        comp.calculate_derivatives_thermal(net, branch_pit, node_pit, branch_lookups, options)
+        comp.adaption_before_derivatives_thermal(
+            net, branch_pit, node_pit, branch_lookups, options)
+    calculate_derivatives_thermal(net, branch_pit, node_pit, options)
+    for comp in net['component_list']:
+        comp.adaption_after_derivatives_thermal(
+            net, branch_pit, node_pit, branch_lookups, options)
```

Comment:
Just for my understanding: You basically applied the same process as in calculate_hydraulics here right? Does the reduction of the function call to once instead of iterating through components improve calculation speed or is it just for clarity and more organised structure?

Reply:
I would say it does both. By applying the same procedure as in calculate_hydraulics you increase the readability. But it also increases the performance, by just calling this function once instead of calling it in each component seperately. The reason we can do so is, that the general function remains the same. Component specific changes can be done in ..._before_... and ..._after_... but they should be minor, that's why this is a performance boost.